### PR TITLE
Packages: Create new transform package

### PR DESCRIPTION
**Pull request by mohdsayed**
## Context

See https://github.com/GoogleForCreators/web-stories-wp/issues/4737

## Summary

Create a new `transform` package from `story-editor/src/components/transform`.

## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Partially addresses [#456](https://github.com/GoogleForCreators/web-stories-wp/issues/4737)

### CODE REVIEW

File: phpcs.xml.dist [hunk 1]
```diff
@@ -88,7 +88,8 @@
 	</rule>
 
 	<rule ref="WordPress.NamingConventions.ValidFunctionName.MethodNameInvalid">
-		<exclude-pattern>tests/phpunit/integration/includes/REST_Setup.php</exclude-pattern>
+		<exclude-pattern
+    >tests/phpunit/integration/includes/REST_Setup.php</exclude-pattern>
```
Comment (swissspidy) [inline phpcs.xml.dist]: ?

Comment (mohdsayed) [reply phpcs.xml.dist]: This got added in a merge conflict, will update.

File: packages/transform/package.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	"author": {
+		"name": "Google",
+		"url": "https://creators.google/"
+	},
+	"license": "Apache-2.0",
+	"keywords": [
+		"web stories"
+	],
+	"homepage": "https://github.com/googleforcreators/web-stories-wp/",
+	"repository": "git+https://github.com/GoogleForCreators/web-stories-wp.git",
```
Comment (swissspidy) [inline packages/transform/package.json]:
```suggestion
	"repository": {
    "type": "git",
    "url": "https://github.com/GoogleForCreators/web-stories-wp.git",
    "directory": "packages/transform"
  },
```

File: packages/transform/package.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
+{
+	"name": "@googleforcreators/transform",
```
Comment (swissspidy) [inline packages/transform/package.json]: This file should use spaces, not tabs.

Seems like we're not properly linting that 🤔 

Maybe `npx prettier --write packages/transform/package.json` fixes it automatically?

Comment (mohdsayed) [reply packages/transform/package.json]: `prettier --write` is already in place for `lint-staged` but  `package.json` was ignored for prettier in https://github.com/GoogleForCreators/web-stories-wp/commit/177ec3bb91c9bbae85b066ee499101f0ec9719d5

Comment (swissspidy) [reply packages/transform/package.json]: Ah, then I‘ll make this a todo for myself to look into for the future!

File: packages/transform/package.json (hunk 2 above)
Comment (mohdsayed) [reply packages/transform/package.json]: 👍

**Review by swissspidy — APPROVED** [review]

**Review by miina — APPROVED** [review]

**Pull request merged**